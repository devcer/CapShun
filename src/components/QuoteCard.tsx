import React from 'react';
import { StyleSheet, Clipboard, ToastAndroid } from 'react-native';
import { FontAwesome } from '@expo/vector-icons';
import { View, Text } from './Themed';

export default function QuoteCard(props: { quote: Quote }) {
  return (
    <View style={styles.container}>
      <Text style={styles.moodText}>Mood: {props.quote.mood}</Text>
      <Text style={styles.tagsText}>Tags: {props.quote.tags.join(', ')}</Text>
      <Text style={styles.quoteText}>{props.quote.quote}</Text>
      <FontAwesome.Button
        name="copy"
        backgroundColor="#000"
        onPress={copyToClipboard}
        style={styles.icon}
      >
        {/* Copy */}{' '}
        {/* COMMENTED OUT BY GITHUB USER: https://github.com/hambali999, as described by owner ' No need of text Copy.' */}
      </FontAwesome.Button>
    </View>
  );
  function copyToClipboard() {
    Clipboard.setString(props.quote.quote);
    showToast('Copied to clipboard');
  }
}

const styles = StyleSheet.create({
  container: {
    borderColor: '#cccccc',
    borderWidth: 2,
    // flex: 1
  },
  moodText: {
    fontSize: 16,
    padding: 5,
  },
  tagsText: {
    fontSize: 12,
    padding: 5,
  },
  quoteText: {
    fontSize: 20,
    padding: 5,
  },
  copyButton: {},
  icon: {
    alignSelf: 'flex-end',
    // BY GITHUB USER: https:github.com/hambali999, as described by owner to ' Copy icon should be on the bottom right of the card. '
  },
});

export const showToast = (message: string) => {
  ToastAndroid.showWithGravity(
    message,
    ToastAndroid.SHORT,
    ToastAndroid.BOTTOM,
  );
};

export interface Quote {
  id: string;
  mood: string;
  tags: string[];
  quote: string;
}
