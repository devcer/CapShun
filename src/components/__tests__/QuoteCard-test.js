import * as React from 'react';
import renderer from 'react-test-renderer';
import { captions } from '../../constants/MockData';

import QuoteCard from '../QuoteCard';
describe('<QuoteCard />', () => {
  it('has 4 children', () => {
    var tree;
    captions.forEach((item) => {
      tree = renderer.create(<QuoteCard key={item.id} quote={item} />).toJSON();
      expect(tree.children.length).toBe(4);
    });
  });
  test('renders correctly', () => {
    let item = captions[0];
    let tree = renderer
      .create(<QuoteCard key={item.id} quote={item} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
