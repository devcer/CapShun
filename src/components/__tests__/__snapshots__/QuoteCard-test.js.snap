// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuoteCard /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "borderColor": "#cccccc",
        "borderWidth": 2,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 16,
          "padding": 5,
        },
      ]
    }
  >
    Mood: 
    Happy
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 12,
          "padding": 5,
        },
      ]
    }
  >
    Tags: 
    sunset, beach
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 20,
          "padding": 5,
        },
      ]
    }
  >
    Beach please!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#000",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "#000",
            "borderRadius": 5,
          },
          undefined,
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontWeight": "600",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Copy
      </Text>
    </View>
  </View>
</View>
`;
